### PR TITLE
[New Resource]: DataZone Environment Profile

### PR DESCRIPTION
**Pull request by ThomasZalewski**
### Description

This resource will allow practitioners to create DataZone Environment Profiles for a DataZone Domain via Terraform.

### Relations

Relates #37423

### References

https://docs.aws.amazon.com/datazone/latest/APIReference/API_CreateEnvironmentProfile.html

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccDataZoneEnvironmentProfile PKG=datazone
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.5 test ./internal/service/datazone/... -v -count 1 -parallel 20 -run='TestAccDataZoneEnvironmentProfile'  -timeout 360m
=== RUN   TestAccDataZoneEnvironmentProfile_basic
=== PAUSE TestAccDataZoneEnvironmentProfile_basic
=== RUN   TestAccDataZoneEnvironmentProfile_disappears
=== PAUSE TestAccDataZoneEnvironmentProfile_disappears
=== CONT  TestAccDataZoneEnvironmentProfile_basic
=== CONT  TestAccDataZoneEnvironmentProfile_disappears
--- PASS: TestAccDataZoneEnvironmentProfile_disappears (28.81s)
--- PASS: TestAccDataZoneEnvironmentProfile_basic (40.73s)
PASS
...
```

### CODE REVIEW

File: internal/service/datazone/environment_profile.go [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+	state.AwsAccountRegion = flex.StringToFramework(ctx, out.AwsAccountRegion)
+	state.CreatedAt = flex.TimeToFramework(ctx, out.CreatedAt)
+	state.CreatedBy = flex.StringToFramework(ctx, out.CreatedBy)
+	state.Description = flex.StringToFramework(ctx, out.Description)
+	state.DomainIdentifier = flex.StringToFramework(ctx, out.DomainId)
+	state.EnvironmentBlueprintIdentifier = flex.StringToFramework(ctx, out.EnvironmentBlueprintId)
+	state.Id = flex.StringToFramework(ctx, out.Id)
+	state.Name = flex.StringToFramework(ctx, out.Name)
+	state.ProjectIdentifier = flex.StringToFramework(ctx, out.ProjectId)
+	state.UpdatedAt = flex.TimeToFramework(ctx, out.UpdatedAt)
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:229]: `flex.Flatten` similar to the end of `Create`

File: internal/service/datazone/environment_profile.go [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+	conn := r.Meta().DataZoneClient(ctx)
+
+	var plan, state dEnvProfile
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if plan.EnvironmentBlueprintIdentifier.Equal(state.EnvironmentBlueprintIdentifier) &&
+		plan.ProjectIdentifier.Equal(state.ProjectIdentifier) && plan.DomainIdentifier.Equal(state.DomainIdentifier) {
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:247]: Should this be inverted?

```suggestion
	if !plan.EnvironmentBlueprintIdentifier.Equal(state.EnvironmentBlueprintIdentifier) ||
		!plan.ProjectIdentifier.Equal(state.ProjectIdentifier) || !plan.DomainIdentifier.Equal(state.DomainIdentifier) {
```

File: internal/service/datazone/environment_profile.go [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+		state.AwsAccountRegion = flex.StringToFramework(ctx, out.AwsAccountRegion)
+		state.CreatedAt = flex.TimeToFramework(ctx, out.CreatedAt)
+		state.CreatedBy = flex.StringToFramework(ctx, out.CreatedBy)
+		state.Description = flex.StringToFramework(ctx, out.Description)
+		state.DomainIdentifier = flex.StringToFramework(ctx, out.DomainId)
+		state.EnvironmentBlueprintIdentifier = flex.StringToFramework(ctx, out.EnvironmentBlueprintId)
+		state.Id = flex.StringToFramework(ctx, out.Id)
+		state.Name = flex.StringToFramework(ctx, out.Name)
+		state.ProjectIdentifier = flex.StringToFramework(ctx, out.ProjectId)
+		state.UpdatedAt = flex.TimeToFramework(ctx, out.UpdatedAt)
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:273]: `flex.Flatten`

File: internal/service/datazone/environment_profile.go [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.ValueString(), err),
+			err.Error(),
+		)
+		return
+	}
+}
+
+func (r *resourceEnvironmentProfile) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.Split(req.ID, ":")
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:302]: For new resources, multi-part keys should be comma-delimited. Ref: https://github.com/hashicorp/terraform-provider-aws/issues/27843

File: internal/service/datazone/environment_profile.go [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+
+	if len(parts) != 4 {
+		resp.Diagnostics.AddError("Resource Import Invalid ID", fmt.Sprintf(`Unexpected format for import ID (%s), use: "DomainIdentifier:Id"`, req.ID))
+	}
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:306]: This error message indicates two parts, might just need to be updated to include the blueprint and project identifiers as well?

File: internal/service/datazone/environment_profile.go [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+
+	if out == nil {
+		return nil, tfresource.NewEmptyResultError(in)
+	}
+
+	return out, nil
+}
+
+type dEnvProfile struct {
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:338]: The typical naming convention for this struct is `<resourceName>Data` (ex. `environmentProfileData`)

File: internal/service/datazone/environment_profile.go [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+	ProjectIdentifier types.String `tfsdk:"project_identifier"`
+
+	// computed
+	CreatedAt        timetypes.RFC3339 `tfsdk:"created_at"`
+	CreatedBy        types.String      `tfsdk:"created_by"`
+	DomainIdentifier types.String      `tfsdk:"domain_identifier"`
+	UpdatedAt        timetypes.RFC3339 `tfsdk:"updated_at"`
+}
+
+type dUserParameters struct {
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:357]: `userParametersData` would be the convention here.

File: internal/service/datazone/environment_profile_test.go [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
+	}
+
+	var environmentprofile datazone.GetEnvironmentProfileOutput
+	epName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	dName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	pName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+
+	envProfName := "aws_datazone_environment_profile.test"
```
Comment (jar-b) [inline internal/service/datazone/environment_profile_test.go:38]: Typically the resource being tested is referenced as `resourceName` here.

```suggestion
	resourceName := "aws_datazone_environment_profile.test"
```

File: internal/service/datazone/environment_profile_test.go [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+			t := rs.Primary.Attributes["domain_identifier"]
+
+			_, err := conn.GetEnvironmentProfile(ctx, &datazone.GetEnvironmentProfileInput{
+				DomainIdentifier: &t,
+				Identifier:       aws.String(rs.Primary.ID),
+			})
+
+			if errs.IsA[*types.ResourceNotFoundException](err) || errs.IsA[*types.AccessDeniedException](err) {
+				return nil
+			}
```
Comment (jar-b) [inline internal/service/datazone/environment_profile_test.go:164]: The finder function can be re-used here.

File: internal/service/datazone/environment_profile_test.go [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+		if rs.Primary.ID == "" {
+			return create.Error(names.DataZone, create.ErrActionCheckingExistence, tfdatazone.ResNameEnvironmentProfile, name, errors.New("not set"))
+		}
+
+		t := rs.Primary.Attributes["domain_identifier"]
+		conn := acctest.Provider.Meta().(*conns.AWSClient).DataZoneClient(ctx)
+		resp, err := conn.GetEnvironmentProfile(ctx, &datazone.GetEnvironmentProfileInput{
+			DomainIdentifier: &t,
+			Identifier:       &rs.Primary.ID,
+		})
```
Comment (jar-b) [inline internal/service/datazone/environment_profile_test.go:192]: Re-use finder

File: internal/service/datazone/environment_profile_test.go [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+	_, err := conn.ListEnvironmentProfiles(ctx, input)
+
+	if acctest.PreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+	if err != nil {
+		t.Fatalf("unexpected PreCheck error: %s", err)
+	}
+}
+*/
```
Comment (jar-b) [inline internal/service/datazone/environment_profile_test.go:218]:
```suggestion
```

File: internal/service/datazone/environment_profile_test.go [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+
+func testAccAuthorizerEnvProfImportStateIdFunc(resourceName string) resource.ImportStateIdFunc {
+	return func(s *terraform.State) (string, error) {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return "", fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		//return fmt.Sprintf("%s:%s", (rs.Primary.Attributes["domain_identifier"] , rs.Primary.ID, rs.Primary.Attributes["environment_blueprint_identifier"], rs.Primary.Attributes["project_identifier"])), nil
```
Comment (jar-b) [inline internal/service/datazone/environment_profile_test.go:227]:
```suggestion
```

File: internal/service/datazone/environment_profile_test.go [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+}
+	`))
+}
+
+func testAccEnvironmentProfileConfig_basic(rName, domainName, projectName string) string {
+	return acctest.ConfigCompose(testAccEnvironmentProfileConfig_base(domainName, projectName), fmt.Sprintf(`
+
+
+
+
```
Comment (jar-b) [inline internal/service/datazone/environment_profile_test.go:258]: Remove whitespace for readability.
 
```suggestion
```

File: internal/service/datazone/environment_profile_test.go [hunk 14]
```diff
@@ -0,0 +1,296 @@
...
+}
+`, rName))
+}
+
+func testAccEnvironmentProfileConfig_update(rName, domainName, projectName string) string {
+	return acctest.ConfigCompose(testAccEnvironmentProfileConfig_base(domainName, projectName), fmt.Sprintf(`
+
+
+
+
```
Comment (jar-b) [inline internal/service/datazone/environment_profile_test.go:281]:
```suggestion
```

File: website/docs/r/datazone_environment_profile.html.markdown [hunk 15]
```diff
@@ -0,0 +1,153 @@
...
+
+This resource exports the following attributes in addition to the arguments above:
+
+* `created_at` - Creation time of environment profile.
+* `created_by` - Creator of environment profile.
+* `id` - ID of environment profile.
+* `updated_at` - Time of last update to environment profile.
+
+## Timeouts
+
```
Comment (jar-b) [inline website/docs/r/datazone_environment_profile.html.markdown:137]:
```suggestion
```

File: website/docs/r/datazone_environment_profile.html.markdown [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+## Import
+
+In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import DataZone Environment Profile using the `example_id_arg`. For example:
```
Comment (jar-b) [inline website/docs/r/datazone_environment_profile.html.markdown:140]: For multi-part keys we typically state which arguments are used and how they are delimited.

**Review by jar-b — COMMENTED** [review]
Looking good 👍 . Just a few suggestions around naming and data handling standards.

File: internal/service/datazone/environment_profile.go [hunk 17]
```diff
@@ -0,0 +1,361 @@
...
+
+func (r *resourceEnvironmentProfile) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			names.AttrAWSAccountID: schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexache.MustCompile("^\\d{12}$"), "must match ^\\d{12}$"),
+				},
```
[the remark this answers is not in the document]
Comment (ThomasZalewski) [reply internal/service/datazone/environment_profile.go:60]: Thanks, that makes sense and I'll make sure to do that in the future

File: website/docs/r/datazone_environment_profile.html.markdown (hunk 16 above)
Comment (ThomasZalewski) [reply website/docs/r/datazone_environment_profile.html.markdown]:
```terraform
import {
  to = aws_datazone_environment_profile.example
  id = "domain_identifier:environment_profile_id:environment_blueprint_identifier:project_identifier"
}
```

Using `terraform import`, import DataZone Environment Profile using the `id`. For example:

```console
% terraform import aws_datazone_environment_profile.example environment_profile-id-12345678
```

Would this work?

File: internal/service/datazone/environment_profile.go [hunk 18]
```diff
@@ -0,0 +1,361 @@
...
+	}
+
+	in := &datazone.CreateEnvironmentProfileInput{}
+	resp.Diagnostics.Append(flex.Expand(ctx, &plan, in)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	in.Name = plan.Name.ValueStringPointer()
+	in.EnvironmentBlueprintIdentifier = plan.EnvironmentBlueprintIdentifier.ValueStringPointer()
+	in.AwsAccountId = plan.AwsAccountId.ValueStringPointer()
```
[the remark this answers is not in the document]
Comment (ThomasZalewski) [reply internal/service/datazone/environment_profile.go:150]: flex.expand is really inconsistent for me

File: website/docs/r/datazone_environment_profile.html.markdown (hunk 16 above)
Comment (jar-b) [reply website/docs/r/datazone_environment_profile.html.markdown]: Here's an example of a recent resource where the ID is a multi-part key (and therefore has no `id` attribute):
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_runtime_management_config#import

This resource can take the same approach and drop the computed `id` attribute as it's a redundant combination of other existing attributes.

File: internal/service/datazone/environment_profile_test.go [hunk 19]
```diff
@@ -0,0 +1,328 @@
...
+  managed   = true
+}
+
+resource "aws_datazone_environment_blueprint_configuration" "test" {
+  domain_id                = aws_datazone_domain.test.id
+  environment_blueprint_id = data.aws_datazone_environment_blueprint.test.id
+  provisioning_role_arn    = aws_iam_role.domain_execution_role.arn
+  enabled_regions          = [data.aws_region.test.name]
+}
+	`))
```
Comment (jar-b) [inline internal/service/datazone/environment_profile_test.go:276]:
```suggestion
`))
```

File: website/docs/r/datazone_environment_profile.html.markdown [hunk 20]
```diff
@@ -0,0 +1,150 @@
...
+This resource exports the following attributes in addition to the arguments above:
+
+* `created_at` - Creation time of environment profile.
+* `created_by` - Creator of environment profile.
+* `id` - ID of environment profile.
+* `updated_at` - Time of last update to environment profile.
+
+## Import
+
+In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import DataZone Environment Profile using the `id`. For example:
```
Comment (jar-b) [inline website/docs/r/datazone_environment_profile.html.markdown:137]:
```suggestion
In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import DataZone Environment Profile using a comma-delimited string combining `domain_identifier` and `id`. For example:
```

File: website/docs/r/datazone_environment_profile.html.markdown [hunk 21]
```diff
@@ -0,0 +1,150 @@
...
+
+```terraform
+import {
+  to = aws_datazone_environment_profile.example
+  id = "domain-id-12345678,environment_profile-id-12345678"
+}
+```
+
+Using `terraform import`, import DataZone Environment Profile using a comma-delimited string combining `environment-profile-id` and `domain-id`. For example:
```
Comment (jar-b) [inline website/docs/r/datazone_environment_profile.html.markdown:146]:
```suggestion
Using `terraform import`, import DataZone Environment Profile using a comma-delimited string combining `domain_identifier` and `id`. For example:
```

File: website/docs/r/datazone_environment_profile.html.markdown [hunk 22]
```diff
@@ -0,0 +1,150 @@
...
+
+```console
+% terraform import aws_datazone_environment_profile.example environment-domain-id-12345678,environment_profile-id-12345678
```
Comment (jar-b) [inline website/docs/r/datazone_environment_profile.html.markdown:149]:
```suggestion
% terraform import aws_datazone_environment_profile.example domain-id-12345678,environment_profile-id-12345678
```

File: website/docs/r/datazone_environment_profile.html.markdown [hunk 23]
```diff
@@ -0,0 +1,150 @@
...
+* `aws_account_Id` - (Required) -  Id of the AWS account being used. Must follow regex of ^\d{12}$.
+* `aws_account_region` - (Required) -  Desired region for environment profile. Must follow regex of ^[a-z]{2}-[a-z]{4,10}-\d$.
+* `domain_identifier` - (Required) -  Domain Identifier for environment profile.
+* `name` - (Required) -  Name of the environment profile. Must follow regex of ^[\w -]+$ and have the length between 1 and 64.
+* `environment_blueprint_identifier` - (Required) -  ID of the blueprint which the environment will be created with. Must follow regex of ^[a-zA-Z0-9_-]{1,36}$.
+* `project_identifier` - (Required) -  Project identifier for environment profile. Must follow regex of ^[a-zA-Z0-9_-]{1,36}$.
+
+The following arguments are optional:
+
+* `description` - (Optional) Description of environment profile. Must be between the length of 0 and 2048.
```
Comment (jar-b) [inline website/docs/r/datazone_environment_profile.html.markdown:121]: Regex and string length requirements are typically omitted from these descriptions. If we choose to keep them it'd help readability to wrap the regex text in backticks (eg. `^\d{12}$`).

File: website/docs/r/datazone_environment_profile.html.markdown [hunk 24]
```diff
@@ -0,0 +1,150 @@
...
+    value = "value"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are required:
+
+* `aws_account_Id` - (Required) -  Id of the AWS account being used. Must follow regex of ^\d{12}$.
```
Comment (jar-b) [inline website/docs/r/datazone_environment_profile.html.markdown:112]:
```suggestion
* `aws_account_id` - (Required) -  Id of the AWS account being used. Must follow regex of ^\d{12}$.
```

File: internal/service/datazone/environment_profile.go [hunk 25]
```diff
@@ -0,0 +1,311 @@
...
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.String(), err),
+			err.Error(),
+		)
+		return
+	}
+
+	option := flex.WithIgnoredFieldNames([]string{"UserParameters"})
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:194]:
```suggestion
	option := flex.WithIgnoredFieldNamesAppend("UserParameters")
```

**Review by jar-b — COMMENTED** [review]
Looking good! Just a few suggestions on AutoFlex options and clarifying the import logic and documentation.

File: internal/service/datazone/environment_profile.go [hunk 26]
```diff
@@ -0,0 +1,311 @@
...
+	}
+
+	out, err := findEnvironmentProfileByID(ctx, conn, state.Id.ValueString(), state.DomainIdentifier.ValueString())
+	if tfresource.NotFound(err) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.String(), err),
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:188]:
```suggestion
			create.ProblemStandardMessage(names.DataZone, create.ErrActionReading, ResNameEnvironmentProfile, state.Id.String(), err),
```

File: internal/service/datazone/environment_profile.go [hunk 27]
```diff
@@ -0,0 +1,311 @@
...
+
+		resp.Diagnostics.Append(flex.Expand(ctx, plan, in)...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+		in.Identifier = state.Id.ValueStringPointer()
+		out, err := conn.UpdateEnvironmentProfile(ctx, in)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.ValueString(), err),
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:224]:
```suggestion
				create.ProblemStandardMessage(names.DataZone, create.ErrActionUpdating, ResNameEnvironmentProfile, state.Id.ValueString(), err),
```

File: internal/service/datazone/environment_profile.go [hunk 28]
```diff
@@ -0,0 +1,311 @@
...
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	_, err := conn.DeleteEnvironmentProfile(ctx, &datazone.DeleteEnvironmentProfileInput{
+		DomainIdentifier: state.DomainIdentifier.ValueStringPointer(),
+		Identifier:       state.Id.ValueStringPointer(),
+	})
+
+	if err != nil && !errs.IsA[*awstypes.ResourceNotFoundException](err) {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameEnvironmentProfile, state.Id.ValueString(), err),
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:250]:
```suggestion
			create.ProblemStandardMessage(names.DataZone, create.ErrActionDeleting, ResNameEnvironmentProfile, state.Id.ValueString(), err),
```

File: internal/service/datazone/environment_profile.go [hunk 29]
```diff
@@ -0,0 +1,311 @@
...
+		)
+		return
+	}
+}
+
+func (r *resourceEnvironmentProfile) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.Split(req.ID, ",")
+
+	if len(parts) != 2 {
+		resp.Diagnostics.AddError("Resource Import Invalid ID", fmt.Sprintf(`Unexpected format for import ID (%s), use: "DomainIdentifier,Id,EnvironmentBlueprint,Id,ProjectIdentifier"`, req.ID))
```
Comment (jar-b) [inline internal/service/datazone/environment_profile.go:261]:
```suggestion
		resp.Diagnostics.AddError("Resource Import Invalid ID", fmt.Sprintf(`Unexpected format for import ID (%s), use: "DomainIdentifier,Id"`, req.ID))
```

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
% make testacc PKG=datazone TESTS=TestAccDataZoneEnvironmentProfile_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.6 test ./internal/service/datazone/... -v -count 1 -parallel 20 -run='TestAccDataZoneEnvironmentProfile_'  -timeout 360m

--- PASS: TestAccDataZoneEnvironmentProfile_disappears (30.12s)
--- PASS: TestAccDataZoneEnvironmentProfile_basic (32.06s)
--- PASS: TestAccDataZoneEnvironmentProfile_update (46.75s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/datazone   52.788s
```

**Pull request merged**